lib/domain/database/alert_provider.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:ffi';
+
+import 'package:flutter/material.dart';
+import '../model/alert.dart';
+import 'alertDatabase.dart';
+
+class AlertProvider extends ChangeNotifier {
+  List<Alert> _items = [];
+  List<Alert> get items => _items;
+
+  String _filter = "all";
+  String _sortOrder = 'Ascending';
+  // TODO Kejun: need to improve above
+
+  Future insertDatabase(Alert alert) async {
+    AlarmDatabase.instance.create(alert);
+    _items.add(alert);
+
+    notifyListeners();
+  }
+
+  // Future<void> retrieveAlerts() async {
+  //   retrieveAlerts(true);
+  // }
+
+  Future<void> retrieveAlerts({bool shouldNotify = true}) async {
+    final dataList = await AlarmDatabase.instance.readAllAlerts();
+    _items = dataList;
+
+    if (shouldNotify) {
+      notifyListeners();
+    }
+  }
+
+  // TODO kejun: filter vs sort?? use enum instead of string
+  Future<void> setFilter(String filter) async {
```

Comment:
The problem of current approach is that if the user set `Descending` order, then set `pending` filter, the order would be reverted to default. 
I think we should still deal with sorting and status filter (and potentially more filers) separately. Probably refactor this block to something like
```
  Future<void> setFilter(String? status, String? order, String[]? tags) async {
     if (status != null && status != _status) {
       // case switch new status
     }
     if (order!= null && order!= _order) {
       // case switch new sort order
     }
     if (tags!= null &&  tags != _tags) {
       // case switch new custom tags

     }
  }
```

Reply:
yea right :D Sorry I didn't notice that the sort is already provided in AlertFilterBar. I thought everything in the callback is "filter". I have now fixed it in the new commit.